gax/src/main/java/com/google/api/gax/batching/DynamicFlowControlSettings.java

```diff
@@ -116,9 +116,10 @@ public DynamicFlowControlSettings build() {
 
     private void verifyElementCountSettings(DynamicFlowControlSettings settings) {
       boolean isEnabled =
-          settings.getInitialOutstandingElementCount() != null
-              || settings.getMinOutstandingElementCount() != null
-              || settings.getMaxOutstandingElementCount() != null;
+          settings.getLimitExceededBehavior() != LimitExceededBehavior.Ignore
```

Comment:
can you split this out to a separate early return if statement? 

// disabled by behavior
if (behavior == ignore) {
  return;
}

// disabled by uniform null counts
if (initialcount != null || mincount != null || macmount != null) {
  return;
}

same for below

Reply:
+1